Handle no-drop zones

w5922xd opened the pull request:
SMR player Target mentioned that MGU was not working for newer games because of No-Drop Zones. 

In the sector file, the inclusion of the No-Drop Zone location will crash the application when opening the game universe. MGU parses the `.smr` file by to set the location name and type. If no location type is included then an error occurs when drawing sectors. The `No-Drop Zone` location has no type specified in the `.smr` files. 

This adds support for No-Drop Zones and future locations that may not have a type by simply setting the location type to the name of the location.

I've also added the NDZ icon from the game into the location icons so it will display in the MGU map.

hemberger commented on MGU/Classes/SMR16.cs:
```diff
@@ -356,6 +356,13 @@ public static bool LoadData(string SMRFile, Game currentGame)
+                                if (locData.Length == 1)
+                                {
+                                    currentGame.location[locnr].location_type = locData[0];
+                                }
```
I was debating between ignoring the NDZ in MGU or fixing the .smr file to insist that every location has a "type", but I like this solution more!

hemberger commented on MGU/Classes/SMR16.cs:
```diff
@@ -393,6 +400,10 @@ public static bool LoadData(string SMRFile, Game currentGame)
+                                        case "No-Drop Zone":
+                                            // In case No-drop Zone location type is added to smr files.
+                                            currentGame.location[locnr].location_type = locData[0];                                                                                    
+                                            break;
```
I think we could probably remove this. I don't expect to give the NDZ a type named after itself.

w5922xd replied:
Sounds good, removed!

hemberger commented on MGU/Forms/MainStuff.cs:
```diff
@@ -629,6 +629,7 @@ private void InitializeComponent()
             this.Locs.Images.SetKeyName(10, "");
             this.Locs.Images.SetKeyName(11, "enemy_forces.GIF");
             this.Locs.Images.SetKeyName(12, "friendly_forces.GIF");
+            this.Locs.Images.SetKeyName(13, "ndz.png");
```
Gosh, `.resx` files are an awful way to store data in git repos. There's really no way to know what it could contain. I'm assuming the changes just add `ndz.png`, but I'd be happier if we could commit them as separate files.

w5922xd replied:
I hadn't ever stored anything in `.resx` before but wanted to follow the convention in MGU. Completely agree with your assessment though.

I'm not sure this inclusion of `ndz.png` displayed on sectors adds a great deal of utility. In my case I wouldn't find much value in it but I don't know how others use MGU. 

Do you think we should just revert it and not worry about displaying the NDZ icon on sectors?

<img width="574" height="412" alt="image" src="https://github.com/user-attachments/assets/ff28ebfc-c700-4c3c-908d-56319234b876" />

hemberger reviewed (changes requested):
Thanks @w5922xd! It's actually just a warning, not an error. All routes are still generated properly. That said, definitely still worth fixing to avoid this type of confusion.

w5922xd commented:
> Thanks @w5922xd! It's actually just a warning, not an error. All routes are still generated properly. That said, definitely still worth fixing to avoid this type of confusion.

I think Target and I both didn't realize it was a warning. His workaround was to remove all references to NDZ in the smr files. I see now you can hit Continue and still retain most functionality. 

<img width="925" height="605" alt="image" src="https://github.com/user-attachments/assets/ca921684-b8cd-46a1-a67e-06bac744c9db" />

Commit:
checking for valid location icon

The problem with ndz location types in the `.smr` file was trying to access an icon at index `-1 ` in the images array for an unknown location type. This adds a guard to check if the icon exists first.

w5922xd commented:
@hemberger  - I remembered this tonight. From your feedback on avoiding NDZ as a location type and the way images are stored, it made sense to not add any git history with it. Reverted those changes. 

The problem with ndz location types in the `.smr` file was an error while attempting to draw the location icon at image array index  `-1 `. It would also be possible to have an image index for a nonexistent image. This adds a guard to check if the icon exists first.
